helm-v3/005-test.md

```diff
@@ -0,0 +1,27 @@
+# Test Command
+
+## Overview
+
+`helm test` gives a chart author the ability to define actions on a helm release which validate that the chart is running as intended. This is useful for ensuring that a `helm install` or `helm upgrade` created/modified Kubernetes resources successfully and that the consequential `helm release` is functional.
+* A **test** is defined as a command and some arguments executed from a container running in a Kubernetes Pod. If the command completes inside the container with an exit code of 0, it is successful. All other exit codes indicate the test failed.
+* Tests are defined in Manifests that live inside of the `templates/tests/success/` and `templates/tests/failure/` directories inside of a chart.
+  * Tests that live in `templates/tests/success/` are expected to succeed.
+  * Tests that live in `templates/tests/failure` are expected to fail.
```

Comment:
If we are to make a breaking change, I'd like to eliminate failure test cases. In practice they are hardly used -- on github it appears that there are about [20 usages of test-failure](https://github.com/search?l=YAML&q=helm.sh%2Fhook+test-failure&type=Code) and [8000 usages of test-success](https://github.com/search?l=YAML&q=helm.sh%2Fhook+test-success&type=Code) (0 and 34 respectively in helm/charts).

Where `test-failure` _is_ used it can be replaced by a small amount of logic in the test itself (and this is a better fit since there are a variety of ways a pod can fail aside from what runs in its entrypoint).

Reply:
hey @jlegrone this is an outdated proposal. sorry would you mind if we moved this convo to #90 ?